src/components/core/mixins/form-associated-checkbox-mixin.ts

```diff
@@ -80,6 +80,7 @@ export const SbbFormAssociatedCheckboxMixin = <T extends Constructor<LitElement>
         this._attributeMutationBlocked = true;
       }
       this._checked = Boolean(value);
+      this.toggleAttribute('data-checked', this._checked);
```

Comment:
Why is this needed? Can we avoid this?

Reply:
selection-panel used `data-checked` for styling, in this case we didn't have anything on the component itself (for standalone use) and since we adopted the logic from the native checkbox the `checked` attribute is not suitable for styling the component. I couldn't come up with anything else, but if you have suggestions let me know 😅

Reply:
check "toggle-check", we solved it with willupdate() there. I think either you should remove it from toggle-check or, like in toggle-check, set data-checked in checkbox-panel. I'd prefer to have it in checkbox-panel.

```ts
  protected override async willUpdate(changedProperties: PropertyValues<this>): Promise<void> {
    super.willUpdate(changedProperties);

    if (changedProperties.has('checked')) {
      this.internals.ariaChecked = `${this.checked}`;
      this.toggleAttribute('data-checked', this.checked);
    }
  }
```